opengrok-indexer/src/main/java/org/opengrok/indexer/history/HistoryGuru.java

```diff
@@ -402,62 +402,73 @@ public boolean hasAnnotation(File file) {
                 continue;
             }
 
-            String path;
-            try {
-                path = file.getCanonicalPath();
+            if (types == null) {
```

Comment:
I'd personally refactor this a bit (I don't like `add(null)` part), for example:
```
private List<Repository> getRepositories(final File file, final List<String> types) {
    if (types == null || types.isEmpty()) {
        return Collections.singletonList(RepositoryFactory.getRepository(file, null));
    }
    return types.map(t -> RepositoryFactory.getRepository(file, t)).collect(Collectors.toList()).
}
```
And then I would iterate over the repositories.